step-02/js/main.js

```diff
@@ -1,239 +1,278 @@
 'use strict';
 
-var startButton = document.getElementById('startButton');
-var callButton = document.getElementById('callButton');
-var hangupButton = document.getElementById('hangupButton');
-callButton.disabled = true;
-hangupButton.disabled = true;
-startButton.onclick = start;
-callButton.onclick = call;
-hangupButton.onclick = hangup;
-
-var startTime;
-var localVideo = document.getElementById('localVideo');
-var remoteVideo = document.getElementById('remoteVideo');
-
-localVideo.addEventListener('loadedmetadata', function() {
-  trace('Local video videoWidth: ' + this.videoWidth +
-    'px,  videoHeight: ' + this.videoHeight + 'px');
-});
-
-remoteVideo.addEventListener('loadedmetadata', function() {
-  trace('Remote video videoWidth: ' + this.videoWidth +
-    'px,  videoHeight: ' + this.videoHeight + 'px');
-});
-
-remoteVideo.onresize = function() {
-  trace('Remote video size changed to ' +
-    remoteVideo.videoWidth + 'x' + remoteVideo.videoHeight);
-  // We'll use the first onresize callback as an indication that video has started
-  // playing out.
-  if (startTime) {
-    var elapsedTime = window.performance.now() - startTime;
-    trace('Setup time: ' + elapsedTime.toFixed(3) + 'ms');
+// Set up media stream constant and parameters.
+
+// On this codelab, you will be streaming video only (video: true).
+const mediaStreamConstraints = {
+  video: true,
+};
+
+// Set up to exchange only video.
+const offerOptions = {
+  offerToReceiveVideo: 1,
+};
+
+// Defines initial start time of the call (defined as connection between peers).
+let startTime = null;
+
+// Define peer connections, streams and video elements.
+const localVideo = document.getElementById('localVideo');
+const remoteVideo = document.getElementById('remoteVideo');
+
+let localStream;
+let remoteStream;
+
+let localPeerConnection;
+let remotePeerConnection;
+
+
+// Define MediaStreams callbacks.
+
+// Handles success by adding the MediaStream to the video element.
+const gotLocalMediaStream = (mediaStream) => {
```

Comment:
This is subjective, but I think the old syntax will be clearer to most devs and doesn't lose anything here.

In other words, `function gotLocalMediaStream(mediaStream) {...}`.

I just checked with a couple of colleagues, and they confirmed my bias :).

Reply:
I agree that arrow functions are not the most clear way for developers who are not familiar with ES6.

Questions:

1) Can we leave them for anonymous ones i.e. `() => { doSomething(); }`? Or do we prefer `function() { doSomething(); }` also for those?

Example:
`    otherPeer.addIceCandidate(newIceCandidate)
      .then(() => { connectionSucceeded(peerConnection); })
      .catch((error) => { connectionFailed(peerConnection, error); });`
vs:
`    otherPeer.addIceCandidate(newIceCandidate)
      .then(function() { connectionSucceeded(peerConnection); })
      .catch(function(error) { connectionFailed(peerConnection, error); });`




2) Would `const gotLocalMediaStream = function(mediaStream) {` work for you? Or are we trying to move away from variable function definitions entirely?

Example:
`const gotLocalMediaStream = function(mediaStream) {`
vs:
function gotLocalMediaStream(mediaStream) {`





For now using these changes.
If still not acceptable, happy to change it to your proposed style.